WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift

```diff
@@ -121,50 +121,48 @@ private extension AnalyticsHubViewModel {
         let currentTimeRange = try timeRangeSelection.unwrapCurrentTimeRange()
         let previousTimeRange = try timeRangeSelection.unwrapPreviousTimeRange()
```

Comment:
If these are the only methods that throw an error in `retrieveData()` now, do we need both of the `catch` statements to handle errors in `updateData()`? It looks like we'll never use the second catch statement that calls `switchToErrorState()`.

I think that's what we want, since we're handling the individual card error states separately, but it means we could remove that extra error handling unless I'm missing something here.

Reply:
Good point! I agree that the second catch clause is unreachable. But we can't have fully typed error handling in Swift, so if we remove it, we get an error:

> Errors thrown from here are not handled because the enclosing catch is not exhaustive

That's a design decision on Swift side. I've seen an argument that a typed throw may prevent API evolution - for example, if you want to add additional possible error type, you'll need to manually update all callers of that API. When we have untyped `catch` it serves as a default handler just in case something else arise.

We can go other way - repurpose untyped `catch` clause only for `TimeRangeGeneratorError`, but that feels as decreased code readability to me + it adds overhead if we later decide to handle more errors.

I'll merge what we have for now, but I'm very open for additional ideas and discussions, since we are introducing more and more throwing async/await methods!